### ISSUE TITLE
Ignore a block or region / Ignore code-blocks on markdown

### ISSUE THREAD

**Issue by davidlaym** [Jason-Rev/vscode-spell-checker#202]
This is maybe related to #150 and #116 

I'm using cspell primarily to write on markdown files, but i'm finding that code examples are getting me a lot of false positives:

![https://i.imgur.com/Na3cfZJ.png](https://i.imgur.com/Na3cfZJ.png)

and I see two possible solutions:

Implement eslint style comments to ignore blocks, like `<!-- cslint:disable -->` and `<!-- cslint:enable -->` 

Or detect the code blocks in markdown and activate the "only comments and string" mode for that block while keeping the all-checked for the rest of the file.

This is a great tool and I'ts making my life easier. Consider adding a donations button or something :)

**Comment by Jason3S** [comment]
@davidlaym Thank you for the feedback.

There are two ways to do this.

1. Do exactly what you suggest, add `<!--- cslint:disable -->` and `<!--- cslint:enable -->` (Note the triple dash) to your `.md` file. This already works as noted in the [README](https://github.com/Jason-Rev/vscode-spell-checker/blob/master/client/README.md#excluding-and-including-text-to-be-checked) See: [Raw README](https://raw.githubusercontent.com/Jason-Rev/vscode-spell-checker/master/client/README.md) and search for `:disable` for examples.
2. Create a regular expression rule that will ignore everything between the back ticks. It is not well documented:

  ```javascript
"cSpell.languageSettings": [
        {
            // use with Markdown files
            "languageId": "markdown",
            // Exclude code.
            "ignoreRegExpList": [
                "/^\\s*```(.|\\n)+?^\\s*```/gm"
            ]
        }
    ]
  ```

**Comment by davidlaym** [comment]
Thanks! That triple dash was my problem. Thanks for pointing it up!!

**Comment by Jason3S** [comment]
I'm closing this for now.

**Comment by noraj** [comment]
@Jason3S Why cSpell is not ignoring code blocks by default or why there is not an option to disable code blocks and inline code ? I use vscode in a lot of different machine and I don't want to copy this in all of them:

```json
"cSpell.languageSettings": [
      {
          // use with Markdown files
          "languageId": "markdown",
          // Exclude code.
          "ignoreRegExpList": [
              "/^\\s*```(.|\\n)+?^\\s*```/gm"
          ]
      }
  ]
```
